src/test/run-make-fulldeps/foreign-exceptions/foo.cpp

```diff
@@ -0,0 +1,44 @@
+#include <assert.h>
+
+struct exception {};
+
+struct drop_check {
+    bool* ok;
+    ~drop_check() {
+        if (ok)
+            *ok = true;
+    }
+};
+
+extern "C" {
+    void rust_catch_callback(void (*cb)());
+
+    static void callback() {
+        throw exception();
+    }
+
+    void throw_cxx_exception() {
+        try {
+            rust_catch_callback(callback);
+            assert(false && "unreachable");
+        } catch (exception e) {
+            return;
+        }
+        assert(false && "did not catch thrown C++ exception");
+    }
+
+    void cxx_catch_callback(void (*cb)(), bool* ok) {
+        drop_check x;
```

Comment:
`x.ok` is uninitialized.
This means it's undefined behavior to call the destructor without the `x.ok = ok;` assignment; which means the C++ compiler could just optimize out `x.ok` and always directly use `ok` in its place -- which might cause the test to succeed even if the `catch (...)` doesn't catch the foreign exception.

Reply:
Oops, it seems I got my C++ initialization rules mixed up.